### PR TITLE
[AUTO] Fix static code scan issue

### PR DESCRIPTION
**Pull request by songbell**
### Details:
 - static scan issues

### Tickets:
 - 117991, 118001, 117990

### CODE REVIEW

File: src/plugins/auto_batch/src/sync_infer_request.cpp [hunk 1]
```diff
@@ -52,7 +52,7 @@ void SyncInferRequest::share_tensors_with_batched_req(const std::set<std::string
         auto batched_tensor = m_batched_request_wrapper->_infer_request_batched->get_tensor(it);
         if (!batched_tensor._so)
             batched_tensor._so = m_batched_request_wrapper->_infer_request_batched._so;
-        res = create_shared_tensor_on_batched_tensor(batched_tensor, name, batched_inputs, m_batch_id, m_batch_size);
+        res = create_shared_tensor_on_batched_tensor(batched_tensor, std::move(name), batched_inputs, m_batch_id, m_batch_size);
```
Comment (riverlijunjie) [inline src/plugins/auto_batch/src/sync_infer_request.cpp:55]: Name is std::string, why it trigger scan issue?

Comment (songbell) [reply src/plugins/auto_batch/src/sync_infer_request.cpp]: string has move constructor and can use here, we are trying to remove potential copy here

**Review by wangleis — APPROVED** [review]

**Comment by wangleis** [comment]
Please create PR for 2023.1 release branch

**Comment by peterchen-intel** [comment]
PR to release branch https://github.com/openvinotoolkit/openvino/pull/19321

**Pull request merged**